File: gems/decomplex/examples/rust/structural-topology.rs
struct Worker;

impl Worker {
    pub fn run(&self, items: Items) {
        self.prepare();
        if true {
            self.validate();
        }
        for item in items {
            self.helper(item);
        }
    }

    fn prepare(&self) {}
    fn ready(&self) -> bool { true }
    pub fn validate(&self) {}
    fn helper(&self, item: Item) { item; }
}

File: gems/decomplex/examples/ruby/structural-topology.rb
# frozen_string_literal: true

class Worker
  def run(items)
    prepare
    if ready()
      validate
    end
    items.each do |item|
      helper(item)
    end
  end

  private

  def prepare; end
  def ready; true; end
  def validate; end
  def helper(item); item; end

  public :validate
end
